### PR TITLE
docs: nested double quotes should be escaped

### PR DESCRIPTION
**Pull request by TennyZhuang**
#### PR Description

In docs, nested double quotes should be escaped.

### CODE REVIEW

File: docs/sources/reference/components/loki/loki.process.md [hunk 1]
```diff
@@ -1284,7 +1284,7 @@ This example replaces the first two instances of the `loki` word by `Loki`:
 ```alloy
 stage.template {
     source   = "output"
-    template = "{{ Replace .Value "loki" "Loki" 2 }}"
+    template = "{{ Replace .Value \"loki\" \"Loki\" 2 }}"
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki/loki.process.md:1287]:
```suggestion
    template = `{{ Replace .Value "loki" "Loki" 2 }}`
```

File: docs/sources/reference/components/loki/loki.process.md [hunk 2]
```diff
@@ -1303,15 +1303,15 @@ Examples:
 ```alloy
 stage.template {
     source   = "output"
-    template = "{{ Trim .Value ",. " }}"
+    template = "{{ Trim .Value \",. \" }}"
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki/loki.process.md:1306]:
```suggestion
    template = `{{ Trim .Value ",. " }}`
```

File: docs/sources/reference/components/loki/loki.process.md [hunk 3]
```diff
@@ -1303,15 +1303,15 @@ Examples:
...
 }
 stage.template {
     source   = "output"
     template = "{{ TrimSpace .Value }}"
 }
 stage.template {
     source   = "output"
-    template = "{{ TrimPrefix .Value "--" }}"
+    template = "{{ TrimPrefix .Value \"--\" }}"
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki/loki.process.md:1314]:
```suggestion
    template = `{{ TrimPrefix .Value "--" }}`
```

File: docs/sources/reference/components/loki/loki.process.md [hunk 4]
```diff
@@ -1329,11 +1329,11 @@ substituted directly, without using Expand.
 ```alloy
 stage.template {
     source   = "output"
-    template = "{{ regexReplaceAll "(a*)bc" .Value "${1}a" }}"
+    template = "{{ regexReplaceAll \"(a*)bc\" .Value \"${1}a\" }}"
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki/loki.process.md:1332]:
```suggestion
    template = `{{ regexReplaceAll "(a*)bc" .Value "${1}a" }}`
```

File: docs/sources/reference/components/loki/loki.process.md [hunk 5]
```diff
@@ -1329,11 +1329,11 @@ substituted directly, without using Expand.
...
 }
 stage.template {
     source   = "output"
-    template = "{{ regexReplaceAllLiteral "(ts=)" .Value "timestamp=" }}"
+    template = "{{ regexReplaceAllLiteral \"(ts=)\" .Value \"timestamp=\" }}"
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki/loki.process.md:1336]:
```suggestion
    template = `{{ regexReplaceAllLiteral "(ts=)" .Value "timestamp=" }}`
```

File: docs/sources/reference/components/loki/loki.process.md [hunk 6]
```diff
@@ -1347,11 +1347,11 @@ Examples:
 ```alloy
 stage.template {
     source   = "output"
-    template = "{{ Hash .Value "salt" }}"
+    template = "{{ Hash .Value \"salt\" }}"
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki/loki.process.md:1350]:
```suggestion
    template = `{{ Hash .Value "salt" }}`
```

File: docs/sources/reference/components/loki/loki.process.md [hunk 7]
```diff
@@ -1347,11 +1347,11 @@ Examples:
...
 }
 stage.template {
     source   = "output"
-    template = "{{ Sha2Hash .Value "salt" }}"
+    template = "{{ Sha2Hash .Value \"salt\" }}"
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki/loki.process.md:1354]:
```suggestion
    template = `{{ Sha2Hash .Value "salt" }}`
```

**Review by clayton-cornell — COMMENTED** [review]
@TennyZhuang Thanks for catching this.

I talked to the development team and they suggested that we could actually use raw strings instead of escaping the double quotes. Refer to: https://grafana.com/docs/alloy/next/get-started/configuration-syntax/expressions/types_and_values/#raw-strings for details on how it works.

I've added suggestions for each of your proposed changes.

**Comment by TennyZhuang** [comment]
> I've added suggestions for each of your proposed changes.

Looks better, thanks!

**Review by clayton-cornell — APPROVED** [review]
Thanks for submitting this PR @TennyZhuang   The changes we've discussed here look great. Approving and merging :-)

**Pull request merged**